registry-ui/src/hooks/useGenerateReport.tsx

```diff
@@ -0,0 +1,82 @@
+// Custom hook that manages the react-query/api calls for the "Export" feature. 
+import { useState } from "react"
+import { useQuery } from "@tanstack/react-query"
+
+import { generateReport } from "react-libs"
+import { ItemSubType } from "provena-interfaces/RegistryModels"
+import { GenerateReportResponse } from "react-libs/provena-interfaces/ProvenanceAPI"
+
+export interface useGenerateReportProps {
+    // What is the Node ID (either model run or study entity)
+    nodeId: string | undefined
+    itemSubType: ItemSubType | undefined
+    // Chosen depth by the user.
+    depth: string
+    onSuccess?: (response: GenerateReportResponse) => void;
+
+}
+
+export const useGenerateReport = (props: useGenerateReportProps) => {
+
+    // Validate that everything is present correctly.
+    const requiredItems = [props.nodeId, props.depth]
+    const dataReady = !requiredItems.some((i) => {
+        return i === undefined;
+    });
+
+    /* onSuccess is a callback that is fired anytime the query 
+       succesfully fetches new data.
+     */ 
+
+    // Setting up the useQuery
+    const generateReportQuery = useQuery<GenerateReportResponse>({
+        queryKey: ['depth-fetch'],
```

Comment:
Okay, yep thought useQuery was the best for this use-case since it was a GET request as we discussed, but since its POST now and we are making some small changes on the server side, I will change it to a UseMutate.